modules/data.atmosphere/R/extract_local_CMIP5.R

```diff
@@ -0,0 +1,299 @@
+##' Extract NLDAS from local download
+##' Extract NLDAS meteorology for a poimt from a local download of the full grid
+# ----------------------------------- 
+# Description
+# -----------------------------------
+##' @title extract.local.CMIP5
+##' @family 
+##' @author Christy Rollinson, 
+##' @description This function extracts CMIP5 data from grids that have been downloaded and stored locally.
+##'              Files are saved as a netCDF file in CF conventions at *DAILY* resolution.  Note: At this point
+##'              in time, variables that are only available at a native monthly resolution will be repeated to
+##'              give a pseudo-daily record (and can get dealt with in the downscaling workflow).  These files 
+##'              are ready to be used in the general PEcAn workflow or fed into the downscaling workflow.
+# ----------------------------------- 
+# Parameters
+# -----------------------------------
+##' @param outfolder - directory where output files will be stored
+##' @param in.path - path to the raw full grids
+##' @param start_date - first day for which you want to extract met (yyyy-mm-dd)
+##' @param end_date - last day for which you want to extract met (yyyy-mm-dd)
+##' @param site_id name to associate with extracted files
+##' @param lat.in site latitude in decimal degrees
+##' @param lon.in site longitude in decimal degrees
+##' @param model which GCM to extract data from
+##' @param scenario which experiment to pull (p1000, historical, ...)
+##' @param ensemble_member which CMIP5 experiment ensemble member
+##' @param overwrite logical. Download a fresh version even if a local file with the same name already exists?
+##' @param verbose logical. to control printing of debug info
+##' @param ... Other arguments, currently ignored
+##' @export
+##' @examples
+# -----------------------------------
+extract.local.CMIP5 <- function(outfolder, in.path, start_date, end_date, site_id, lat.in, lon.in, 
+                                model , scenario , ensemble_member = "r1i1p1",
+                                overwrite = FALSE, verbose = FALSE, ...){
+  library(lubridate)
+  library(ncdf4)
+  library(stringr)
+  
```

Comment:
Library calls shouldn't be needed if each function from these packages is considered throughout (i.e. ncdf4::ncvar_get)